prov/sockets/src/sock_ep.c

```diff
@@ -584,7 +584,10 @@ static ssize_t sock_tx_size_left(struct fid_ep *ep)
 	fastlock_acquire(&tx_ctx->wlock);
 	num_left = rbavail(&tx_ctx->rb)/SOCK_EP_TX_ENTRY_SZ;
 	fastlock_release(&tx_ctx->wlock);
-	return num_left;
+	if (num_left > SOCK_EP_TX_SZ)
```

Comment:
The return is not necessarily power of two. This is how we got 368 before-
SOCK_EP_TX_ENTRY_SZ = 712
SOCK_EP_TX_SZ = 256
buffer size = next_power_of_two(712) * 256 = 1024 * 256 = 262144
size_left = available_buffer_size/SOCK_EP_TX_ENTRY_SZ = 368